### PR TITLE
plan: new plan support union scan.

### PR DESCRIPTION
**Pull request by hanfei1991**
When a txn is open, we always add a union scan on top of tabel scan or index scan.
@shenli @coocood @zimulala @winoros PTAL

### CODE REVIEW

**Comment by shenli** [comment]
Why we should always add a union scan?

File: plan/new_physical_plan_builder.go [hunk 1]
```diff
@@ -189,6 +189,19 @@ func (p *baseLogicalPlan) convert2NewPhysicalPlan(prop *requiredProp) (taskProfi
 	return task, p.storeTaskProfile(prop, task)
 }
 
+func tryToAddUnionScan(cop *copTaskProfile, conds []expression.Expression, ctx context.Context, allocator *idAllocator) taskProfile {
+	if ctx.Txn() == nil || !ctx.Txn().IsReadOnly() {
```
Comment (shenli) [inline plan/new_physical_plan_builder.go]: If the txn is read only, we do not need to add union scan.

Comment (hanfei1991) [inline plan/new_physical_plan_builder.go]: any problem?

Comment (coocood) [inline plan/new_physical_plan_builder.go]: remove the `!`.

File: plan/dag_plan_test.go [hunk 2]
```diff
@@ -120,19 +138,78 @@ func (s *testPlanSuite) TestDAGPlanBuilderSimpleCase(c *C) {
...
+	}
+	for _, tt := range tests {
+		comment := Commentf("for %s", tt.sql)
+		stmt, err := s.ParseOneStmt(tt.sql, "", "")
+		c.Assert(err, IsNil, comment)
+
+		is, err := plan.MockResolve(stmt)
+		c.Assert(err, IsNil)
+
+		err = se.NewTxn()
```
Comment (coocood) [inline plan/dag_plan_test.go]: New transaction without any data written doesn't need union scan.
We should test union scan with and without dirty data.

Comment (hanfei1991) [inline plan/dag_plan_test.go]: fixed

**Comment by coocood** [comment]
LGTM

File: plan/new_physical_plan_builder.go [hunk 3]
```diff
@@ -189,6 +189,19 @@ func (p *baseLogicalPlan) convert2NewPhysicalPlan(prop *requiredProp) (taskProfi
...
+func tryToAddUnionScan(cop *copTaskProfile, conds []expression.Expression, ctx context.Context, allocator *idAllocator) taskProfile {
+	if ctx.Txn() == nil || ctx.Txn().IsReadOnly() {
+		return cop
+	}
+	task := cop.finishTask(ctx, allocator)
+	us := PhysicalUnionScan{
+		Condition: expression.ComposeCNFCondition(ctx, conds...),
+	}.init(allocator, ctx)
+	us.SetSchema(task.plan().Schema())
+	task = us.attach2TaskProfile(task)
```
Comment (shenli) [inline plan/new_physical_plan_builder.go]: return us.attach2TaskProfile(task)

**Comment by hanfei1991** [comment]
@shenli PTAL

**Pull request merged**